### ISSUE TITLE
jenkins: 32-bit compile failure with sanitizers + F08 binding

### ISSUE THREAD

**Issue by raffenet** [pmodels/mpich#5736]
Using our Jenkins Singularity container w/ GCC 9.1.0.

```
src/binding/fortran/use_mpi_f08/pmpi_f08.f90:5568:41:

 5568 |         SUBROUTINE PMPIR_Win_allocate_f08(size, disp_unit, info, comm, baseptr, win, ierror)
      |                                         1
......
 5582 |         SUBROUTINE PMPIR_Win_allocate_f08_large(size, disp_unit, info, comm, baseptr, win, ierror)
      |                                               2
Error: Ambiguous interfaces in generic interface 'pmpi_win_allocate' for ‘pmpir_win_allocate_f08’ at (1) and ‘pmpir_win_allocate_f08_large’ at (2)
src/binding/fortran/use_mpi_f08/pmpi_f08.f90:5598:48:

 5598 |         SUBROUTINE PMPIR_Win_allocate_shared_f08(size, disp_unit, info, comm, baseptr, win, ierror)
      |                                                1
......
 5612 |         SUBROUTINE PMPIR_Win_allocate_shared_f08_large(size, disp_unit, info, comm, baseptr, win, ierror)
      |                                                      2
Error: Ambiguous interfaces in generic interface 'pmpi_win_allocate_shared' for ‘pmpir_win_allocate_shared_f08’ at (1) and ‘pmpir_win_allocate_shared_f08_large’ at (2)
src/binding/fortran/use_mpi_f08/pmpi_f08.f90:5649:41:

 5649 |         SUBROUTINE PMPIR_Win_create_f08ts(base, size, disp_unit, info, comm, win, ierror)
      |                                         1
......
 5662 |         SUBROUTINE PMPIR_Win_create_f08ts_large(base, size, disp_unit, info, comm, win, ierror)
      |                                               2
Error: Ambiguous interfaces in generic interface 'pmpi_win_create' for ‘pmpir_win_create_f08ts’ at (1) and ‘pmpir_win_create_f08ts_large’ at (2)
src/binding/fortran/use_mpi_f08/pmpi_f08.f90:5840:45:

 5840 |         SUBROUTINE PMPIR_Win_shared_query_f08(win, rank, size, disp_unit, baseptr, ierror)
      |                                             1
......
 5853 |         SUBROUTINE PMPIR_Win_shared_query_f08_large(win, rank, size, disp_unit, baseptr, ierror)
      |                                                   2
```

**Comment by raffenet** [comment]
This is in 4.0.x branch. `main` seems to compile fine.

**Comment by hzhou** [comment]
Is there a `diff` of `maint/local_python/binding_f08.py` between `4.0.x` and `main`?

**Comment by raffenet** [comment]
> Is there a `diff` of `maint/local_python/binding_f08.py` between `4.0.x` and `main`?

Nope, doesn't appear to be.

raffenet@mcswl036:main/ (main*) $ git diff origin/4.0.x maint/local_python/binding_f08.py
raffenet@mcswl036:main/ (main*) $

**Comment by hzhou** [comment]
Do you have the link to the failing Jenkins job?

**Comment by raffenet** [comment]
https://pmrs-jenkins.cels.anl.gov/view/mpich-stable-nightly/job/mpich-stable-sanitizers/

**Comment by hzhou** [comment]
Is this for 3.4.x or 4.0.x? From this https://pmrs-jenkins.cels.anl.gov/view/mpich-stable-nightly/job/mpich-stable-sanitizers/compiler=gnu,jenkins_configure=asan,label=centos32,netmod=ch4-ofi/lastSuccessfulBuild/artifact/mpich-stable/c.txt
```
...
configure: creating ./config.status
config.status: creating src/mpid/ch4/shm/posix/posix_eager_array.c
config.status: creating src/mpid/ch4/shm/posix/eager/include/posix_eager_pre.h
config.status: creating modules/hwloc/Makefile
config.status: creating modules/hwloc/include/Makefile
config.status: creating modules/hwloc/hwloc/Makefile
config.status: creating src/mpid/ch4/src/mpid_ch4_net_array.c
config.status: creating src/mpid/ch4/include/netmodpre.h
config.status: creating src/binding/fortran/use_mpi/mpif90model.h
config.status: creating Makefile
config.status: creating examples/Makefile
config.status: creating test/Makefile
config.status: creating test/commands/Makefile
config.status: creating src/include/mpichinfo.h
...
```
It is still doing embedded hwloc configure, which we have separated in 4.0.x.

**Comment by raffenet** [comment]
That's from the last successful build, which is 3.4.3. Check the artifacts from a failed build and it should report 4.0b1 version.

**Comment by hzhou** [comment]
> That's from the last successful build, which is 3.4.3. Check the artifacts from a failed build and it should report 4.0b1 version.

Right :face_with_spiral_eyes:

**Comment by hzhou** [comment]
In config.status:
```
cmd_gen_binding_f08="python3 ./maint/gen_binding_f08.py -fint-size= -aint-size=4 -count-size=8 -cint-size=4" 
```
`fint-size` was missing. Need figure out why...

**Comment by hzhou** [comment]
It is missing this commit https://github.com/pmodels/mpich/pull/5676/commits/6fc06471cd870a74b0b293f2fc53bf50ec408bb1

**Comment by raffenet** [comment]
Thanks, I'll pick it up for the next PR.

**Comment by raffenet** [comment]
Fix included with #5732.
